src/aiai_eval/evaluator.py

```diff
@@ -106,61 +107,96 @@ def evaluate(
-
+        """Prepare the model ID(s) to be evaluated.
         Args:
-            model_id (str or list of str):
-                The model ID(s) of the models to be evaluated.
-
+            model_id (str, list of str or None):
+                The model ID(s) of the models to evaluate. If None then all model IDs
+                will be retrieved.
         Returns:
             sequence of str:
                 The prepared list of model IDs.
         """
-        pass
-
-    def _prepare_dataset_configs(
-        self,
-        dataset: Optional[Union[Sequence[str], str]],
-    ) -> Sequence[DatasetConfig]:
-        """Prepare the task configuration(s) to be benchmarked.
+        model_ids: Sequence[str]
+        if model_id is None:
+            model_ids = self._get_fresh_model_ids(
+                tasks=self.evaluation_config.model_tasks,
+            )
+        elif isinstance(model_id, str):
+            model_ids = [model_id]
+        else:
+            model_ids = model_id
 
-        Args:
-            task (str or list of str):
-                The tasks to evaluate.
+        return model_ids
 
-        Returns:
-            sequence of TaskConfig:
-                The prepared list of task configurations.
-        """
-        pass
-
-    def _benchmark_single(
+    def _evaluate_single(
         self,
-        dataset_config: DatasetConfig,
+        dataset_task: DatasetTask,
         model_id: str,
     ):
-        """Benchmark a single model on a single dataset.
-
+        """Evaluate a single model on a single task.
         Args:
-            dataset_config (DatasetConfig):
-                The dataset configuration to use.
+            dataset_config (DatasetTask):
+                The dataset task configuration to use.
             model_id (str):
                 The model ID to use.
         """
-        pass
+        logger.info(f"Evaluating {model_id} on {dataset_task.pretty_dataset_name}")
+        try:
+            # dataset_obj = self.task_factory.build_dataset(dataset_task)
+            results = (
+                "NA"  # dataset_obj(model_id) # NB NEED TO IMPLEMENT `build_dataset`
+            )
+            self.evaluation_results[dataset_task.name][model_id] = results
+            logger.debug(f"Results:\n{results}")
+
+        except InvalidEvaluation as e:
+
+            # If the model ID is not valid then raise an error, if specified
+            model_err_msg = "does not exist on the Hugging Face Hub"
+            if (
+                self.evaluation_config.raise_error_on_invalid_model
+                and model_err_msg in str(e)
+            ):
+                raise e
+
+            # Otherwise, log the error
+            else:
+                logger.info(
+                    f"{model_id} could not be evaluated on "
+                    f"{dataset_task.pretty_dataset_name}. Skipping."
+                )
+                logger.debug(f'The error message was "{e}".')
```

Comment:
I think we should create a separate `ModelDoesNotExistOnHuggingFaceHubException` here, rather than searching through the error message.